pyproject.toml

```diff
@@ -1,5 +1,172 @@
+eo3-prepare = "eodatasets3.scripts.prepare:run"
+eo3-recompress-tar = "eodatasets3.scripts.recompress:main"
+eo3-package-wagl = "eodatasets3.scripts.packagewagl:run"
+eo3-to-stac = "eodatasets3.scripts.tostac:run"
+
+[project.optional-dependencies]
+# Optional valid-data poly handling methods
+algorithms = ["scikit-image"]
+# The (legacy) prepare scripts
+ancillary = ["checksumdir", "netCDF4"]
+# If packaging ard/wagl.
+wagl = ["h5py"] # h5py is already in core, but list it here for clarity if needed.
+# The original setup.py had 'all' as the union of wagl, ancillary, algorithms
+# And 'test' was defined as the same as 'all'.
+# We need to list the dependencies explicitly here in TOML.
+# Combined dependencies: h5py, checksumdir, netCDF4, scikit-image, deepdiff, mock, pep8-naming, pytest, rio_cogeo, sphinx-autodoc-typehints, sphinx_rtd_theme, pytest-xdist, pytest-cov
+all = [
+    "h5py", # From wagl (and core)
+    "checksumdir", "netCDF4", # From ancillary
+    "scikit-image", # From algorithms
+    # Add test dependencies from setup.py's tests_require, as test was all
+    "deepdiff>=8.0",
+    "mock",
+    "pep8-naming",
+    "pytest",
+    "rio_cogeo",
+    "sphinx-autodoc-typehints",
+    "sphinx_rtd_theme",
+    "pytest-xdist",
+    "pytest-cov",
+]
+# As per setup.py logic: EXTRAS_REQUIRE["test"] = EXTRAS_REQUIRE["all"]
+test = [
+    "h5py", # From wagl (and core)
+    "checksumdir", "netCDF4", # From ancillary
+    "scikit-image", # From algorithms
+    # Add test dependencies from setup.py's tests_require
+    "deepdiff>=8.0",
+    "mock",
+    "pep8-naming",
+    "pytest",
+    "rio_cogeo",
+    "sphinx-autodoc-typehints",
+    "sphinx_rtd_theme",
+    "pytest-xdist",
+    "pytest-cov",
+]
+# Prod deployment just adds the optional wagl depenencies.
+deployment = ["h5py"] # h5py is already in core, but list it here for clarity if needed.
+
+# --- Setuptools configuration ---
+[tool.setuptools]
+# Let setuptools find packages automatically
+# Equivalent to find_packages(exclude=("tests", "tests.*"))
+# packages = ["eodatasets3"] # Specify main package to include
+
+[tool.setuptools.packages.find]
+# where = ["."] # Search in the current directory
+exclude = ["tests", "tests.*"]
+
+[tool.setuptools.package-data]
+# Equivalent to package_data dictionary
+"*" = ["*.json", "*.yaml", "*.db"]
+"eodatasets3" = ["eodatasets3/py.typed"]
+
+
+# --- setuptools-scm configuration (replaces versioneer) ---
+[tool.setuptools_scm]
+# Where to write the version file
+write_to = "eodatasets3/_version.py"
+# Configure versioning based on git tags
+# Matches tags starting with eodatasets3- (like versioneer's tag_prefix)
+tag_regex = "^eodatasets3-(.*)"
+
+# --- Tool Configurations (from setup.cfg) ---
+
+[tool.pytest]
+testpaths = ["eodatasets3", "tests"] # List of paths
+addopts = "--doctest-modules"
+filterwarnings = ["ignore::FutureWarning"] # List of strings
+norecursedirs = [".*", "build", "dist", ".git", "tmp*"] # List of strings
+
+[tool.flake8]
```

Comment:
Same comment about sorting of groups, I would expect to find `tool.flake8` after `tool.coverage` but before `tool.isort`

Reply:
Good idea, I hadn't thought about sorting these TOML sections.